src/icrar/leap-accelerate-cli/main.cc

```diff
@@ -143,28 +145,12 @@ int main(int argc, char** argv)
     {
         ArgumentsValidated args = { Arguments(std::move(rawArgs)) };
 
-        //=========================
-        // Calibration to std::cout
-        //=========================
         LOG(info) << version_information(argv[0]);
         LOG(info) << arg_string(argc, argv);
 
-        switch(args.GetComputeImplementation())
-        {
-        case ComputeImplementation::cpu:
-        {
-            cpu::CalibrateResult result = icrar::cpu::Calibrate(args.GetMeasurementSet(), args.GetDirections(), args.GetMinimumBaselineThreshold(), args.IsFileSystemCacheEnabled());
-            cpu::PrintResult(result, args.GetOutputStream());
-            break;
-        }
-        case ComputeImplementation::cuda:
-        {
-            cpu::CalibrateResult result = icrar::cuda::Calibrate(args.GetMeasurementSet(), args.GetDirections(), args.GetMinimumBaselineThreshold(), args.IsFileSystemCacheEnabled());
-            cpu::PrintResult(result ,args.GetOutputStream());
-            break;
-        }
-        }
-        return 0;
+        auto calibrator = LeapCalibratorFactory().Create(args.GetComputeImplementation());
+        cpu::CalibrateResult result = calibrator->Calibrate(args.GetMeasurementSet(), args.GetDirections(), args.GetMinimumBaselineThreshold(), args.IsFileSystemCacheEnabled());
+        cpu::PrintResult(result, args.GetOutputStream());
```

Comment:
Maybe something shorter like

````c++
auto results = ...;
PrintResult(result, ....);
```

Reply:
this is mostly style preference, in some cases it is preferred to mark the type when it's not easy to infer from the r-value